src/scripts/main.js

```diff
@@ -1,3 +1,300 @@
+  };
+};
+
+function isGameLost() {
+  let result = true;
+
+  [...gameField.rows].forEach(row => {
+    const cells = [...row.cells];
+
+    for (let i = 0; i < cells.length - 1; i++) {
+      const value = cells[i].innerText;
+      const nextValue = cells[i + 1].innerText;
+
+      if ((value === nextValue) || value === '') {
+        result = false;
+
+        return;
+      };
+    };
+  });
+
+  if (result === true) {
+    const columns = [];
+    const fieldRows = [...gameField.rows];
+
+    for (let i = 0; i < fieldRows.length; i++) {
+      const column = [];
+
+      fieldRows.forEach(row => {
+        column.push([...row.cells][i]);
+      });
+
+      columns.push(column);
+    };
+
+    columns.forEach(column => {
+      for (let i = 0; i < column.length - 1; i++) {
+        const value = column[i].innerText;
+        const nextValue = column[i + 1].innerText;
+
+        if ((value === nextValue) || value === '') {
+          result = false;
+
+          return;
+        };
+      };
+    });
+  };
+
+  return result;
+};
+
+function looseGame() {
+  const messageLose = document.querySelector('.message-lose');
+
+  messageLose.classList.remove('hidden');
+};
+
+function winGame() {
+  const messageWin = document.querySelector('.message-win');
+
+  messageWin.classList.remove('hidden');
+};
+
+function startGame() {
+  startButton.classList.remove('start');
+  startButton.style.outline = 'none';
+  startButton.classList.add('restart');
+
+  if (startButton.innerText === 'Start') {
+    startButton.innerText = 'Restart';
+
+    const startMessage = document.querySelector('.message-start');
+
+    startMessage.classList.add('hidden');
+  } else {
+    scoreCounter = 0;
+    scoreInfo.innerText = scoreCounter;
+
+    const cells = getAllCells();
+
+    cells.forEach(cell => {
+      cell.className = 'field-cell';
+      cell.innerText = '';
+    });
+
+    const looseMessage = document.querySelector('.message-lose');
+    const winMessage = document.querySelector('.message-win');
+
+    if (looseMessage) {
+      looseMessage.classList.add('hidden');
+    };
+
+    if (winMessage) {
+      winMessage.classList.add('hidden');
+    }
+  };
+
+  addRandomNumber(getRandomNumber());
+  addRandomNumber(getRandomNumber());
+};
+
+function getAllCells() {
+  return [...document.querySelectorAll('.field-cell')];
+};
+
+function getFreeCells() {
+  return [...document.querySelectorAll('.field-cell')]
+    .filter(cell => cell.innerText === '');
+};
+
+function moveCells(firsCell, secondCell) {
+  const firstValue = Number(firsCell.innerText);
+  const secondValue = Number(secondCell.innerText);
+
+  if (firstValue !== 0 && secondValue === 0) {
+    firsCell.className = 'field-cell';
+    firsCell.innerText = '';
+    secondCell.innerText = firstValue;
+    secondCell.classList.add(`field-cell--${firstValue}`);
+    actionCounter++;
+  };
+
+  if (
+    firstValue !== 0 && firstValue === secondValue
+    && !blockedCells.includes(firsCell)
+  ) {
```

Comment:
move such a long if condition to a variable for better readability and maintainability.